cub/device/dispatch/dispatch_histogram.cuh

```diff
@@ -357,14 +365,12 @@ public:
       }
 
       template <typename T>
-      __host__ __device__ __forceinline__
-      int ComputeBin(T      sample,
-                     T      min_level,
-                     ScaleT scale,
-                     cuda::std::false_type /* is_fp */)
+      __host__ __device__ __forceinline__ int
+      ComputeBin(T sample, T min_level, ScaleT scale, cuda::std::false_type /* is_fp */)
       {
-        return static_cast<int>(((sample - min_level) * scale.fraction.bins) /
-                                scale.fraction.range);
+        return static_cast<int>((static_cast<IntArithmeticT>(sample - min_level) *
+                                 static_cast<IntArithmeticT>(scale.fraction.bins)) /
```

Comment:
Would there be any value in storing `fraction.bins` and `fraction.range` in some combination of `CommonT` / `IntArithmeticT` instead of casting? We might reuse / rename `CommonT` to `ComputeT` and conditionally switch between common type and the integer logic.

Reply:
I've had a similar thought, but had decided against it, because currently `fraction.bins` and `fraction.range` can be packed into a single word with _potentially_ lower register footprint than if we'd store them as `ComputeT`. For `CommonT` of smaller than 32-bit word, those have been promoted anyways. So, the only difference for those types is that, now, we're just explicitly casting those (instead of implicit integer promotion). 

If you feel strongly about it, I'm also ok to change it, though 🙂

Reply:
@elstehle I agree with your arguments. Just thought that we might save a few device-side conversions and a simplify the code a bit. I'm not sure about performance implications here, so it's entirely optional suggestion. Feel free to ignore.